### ISSUE TITLE
Lost my skillpoint

### ISSUE THREAD

**Issue by Zoneman81** [StrangeLoopGames/EcoIssues#6918]
Hello,

During GMT+1 night hours around 4am, i suddenly lost my 15 skillpoint.

Went offline and came back 6 hours later, lost food nutrition points, but gained 0.0 skillpoints.

After 1 hour of playing i starting gaining skillpoints again.

See attached screenshot.
![20180403110645_1](https://user-images.githubusercontent.com/38038590/38247373-5d1bf08a-3745-11e8-9c51-1ab5f8bdd48d.jpg)

**Comment by Kitoka** [comment]
I had the exact same problem on the same server.
![eco_azrael_koteka_sp](https://user-images.githubusercontent.com/38046393/38261437-794e3130-376a-11e8-9198-474ba15df619.jpg)

**Comment by Kitoka** [comment]
I had this happen again multiple times now, but I found out that the SP loss in the messages (screenshots) are not right. I had my SP reset to 0 on my last login to our server while 5 minutes earlier (before the server rebooted) I had around 11SP unspended. The message after logging back in the game stated I lost 61.3SP which was impossible as I only had about 11 SP 5 minutes earlier.

**Comment by Dreepa** [comment]
Just happened: 180 SP... I alt tab out of the game, do some stuff, tab back into the game, 0 SP.
The only thing skill related I did do was I learned advanced cooking. Unfortunately I didn't watch the SP at that moment.

**Comment by craig-jong** [comment]
Are all these reports coming from the same server? Can you check with your server owner if they've run any admin/developer commands recently?

**Comment by Zoneman81** [comment]
I can get you the server logs, if you need those?

 

Fra: craig-jong [mailto:notifications@github.com] 
Sendt: 4. april 2018 20:00
Til: StrangeLoopGames/EcoIssues
Cc: Zoneman81; Author
Emne: Re: [StrangeLoopGames/EcoIssues] Lost my skillpoint (#6918)

 

Are all these reports coming from the same server? Can you check with your server owner if they've run any admin/developer commands recently?

**Comment by Zoneman81** [comment]
The server admin has used the following commands /fly and /give

 

Fra: craig-jong [mailto:notifications@github.com] 
Sendt: 4. april 2018 20:00
Til: StrangeLoopGames/EcoIssues
Cc: Zoneman81; Author
Emne: Re: [StrangeLoopGames/EcoIssues] Lost my skillpoint (#6918)

 

Are all these reports coming from the same server? Can you check with your server owner if they've run any admin/developer commands recently?

**Comment by craig-jong** [comment]
Is it possible you could upload your save file so I can check it?

**Comment by Zoneman81** [comment]
Sure, where is it normally located on my computer?

 

Fra: craig-jong [mailto:notifications@github.com] 
Sendt: 5. april 2018 00:05
Til: StrangeLoopGames/EcoIssues
Cc: Zoneman81; Author
Emne: Re: [StrangeLoopGames/EcoIssues] Lost my skillpoint (#6918)

 

Is it possible you could upload your save file so I can check it?

**Comment by Kitoka** [comment]
Zoneman81, Dreepa and me are playing on the same server.

**Comment by craig-jong** [comment]
Just upload the Game.eco file located in Server/Storage so I can check a few things.

**Comment by Last2Night** [comment]
@craig-jong 
game.eco: https://drive.google.com/open?id=16PV9pnBtrHLzinkcrqLJtZJweo2rSriA

**Comment by Last2Night** [comment]
@craig-jong @MarjieAVolk 
We finally found what causes this bug to appear. 
The world our players played on during the opening of this issue was removed due to all the bugs including this one. Before I start, I want to point our that old world had SP/action enabled at the beginning and disabled shortly after the world was launched. I started a new one and here is the story.

The new world we started was very stable and had minor bugs only until yesterday when we decided to run a labor event. During 1 hour of playtime we enabled SP gain per Action to encourage people to do hard work. During the event everything was fine. When the 1 hour of event passed, I changed SP/action back to 0. Then, people begun to randomly lose their SP. Their accumulated SP was simply reset to zero at random events, on logon, building, crafting, walking even while AFKing. Since We had no issues prior to this event, the first thing crossed my mind - SP/Action somehow corrupted world files. I rolled back to a backup save prior to the event and all SP loss problems were gone. I decided to give it a grace period before posting here, so time frames wise it has been almost 24 hours since we rolled back to backed up save and had no SP losses.

Im pretty sure you will be able to reproduce this bug on your own now.
